locations/pipelines/address_clean_up.py

```diff
@@ -0,0 +1,36 @@
+from scrapy import Spider
+
+from locations.items import Feature
+
+
+def merge_address_lines(address_lines: [str]) -> str:
+    return ", ".join(filter(None, [line.strip() if line else None for line in address_lines]))
+
+
+def clean_address(address: list[str] | str) -> str:
+    if isinstance(address, list):
+        address = merge_address_lines(address)
+
+    address_list = address.replace("\n", ",").replace("\r", ",").replace("\t", ",").replace("\f", ",").split(",")
+
+    return_addr = []
+
+    for line in address_list:
+        if line:
+            line = line.replace("\xa0", " ")
+            line = line.strip("\n\r\t\f ,")
+            if line:
+                return_addr.append(line)
+
+    return ", ".join(return_addr)
+
+
+class AddressCleanUpPipeline:
+    def process_item(self, item: Feature, spider: Spider):
+        if street_address := item.get("street_address"):
+            if isinstance(street_address, str):
+                item["street_address"] = clean_address(street_address)
+        if addr_full := item.get("addr_full"):
+            if isinstance(addr_full, str):
```

Comment:
Should an isinstance(street_address, list) be added here too as an OR condition?